go/ethclient/geth_rpc_client.go

```diff
@@ -157,13 +164,7 @@ func (e *gethRPCClient) BroadcastTx(tx *obscurocommon.L1TxData) {
 	e.lock.Lock()
```

Comment:
I notice this now. What lock is this? Why is it needed?
 Shouldn't it have a more explicit name, and some comments?

Reply:
Let me make the var name more descriptive 👍

Reply:
Yeah nonce is the biggest issue.

Reply:
actually, you don't need CAS. All you need is `atomic.AddInt64`. That guarantees that every tx gets a new nonce.
Now, if a tx fails, tough luck, that nonce is lost.
Am I missing something?

Reply:
Made that change, back to you @tudor-malene  :)